homeassistant/components/script/__init__.py

```diff
@@ -256,6 +256,7 @@ async def service_handler(service):
 
         # Register the service description
         service_desc = {
+            CONF_NAME: cfg.get(CONF_ALIAS, ""),
```

Comment:
We have access to the script entity in this location:

```suggestion
            CONF_NAME: script_entity.name
```